src/embeddings.rs

```diff
@@ -87,8 +88,20 @@ impl PyEmbeddings {
 
         let results =
             match embeddings.analogy_masked(word1, word2, word3, limit, [mask.0, mask.1, mask.2]) {
```

Comment:
Since all the action is in the `Err`, maybe 

```rust
let embeddings = embeddings
  .analogy_masked(word1, word2, word3, limit, [mask.0, mask.1, mask.2])
  .map_err(|oov_words| ...)?;
```

Looks better?